sdk/eventhub/azure-eventhubs/azure/eventhub/aio/eventprocessor/_ownership_manager.py

```diff
@@ -0,0 +1,132 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# -----------------------------------------------------------------------------------
+
+import time
+import random
+import math
+from typing import List
+from collections import Counter, defaultdict
+from azure.eventhub.aio import EventHubClient
+from .partition_manager import PartitionManager
+
+
+class OwnershipManager(object):
+    """Increases or decreases the number of partitions owned by an EventProcessor
+    so the number of owned partitions are balanced among multiple EventProcessors
+
+    An EventProcessor calls claim_ownership() of this class every x seconds,
+    where x is set by keyword argument "polling_interval" in EventProcessor,
+    to claim the ownership of partitions, create tasks for the claimed ownership, and cancel tasks that no longer belong
+    to the claimed ownership.
+
+    """
+    def __init__(
+            self, eventhub_client: EventHubClient, consumer_group_name: str, owner_id: str,
+            partition_manager: PartitionManager, ownership_timeout: int
+    ):
+        self.cached_parition_ids = []  # type: List[str]
+        self.eventhub_client = eventhub_client
+        self.eventhub_name = eventhub_client.eh_name
+        self.consumer_group_name = consumer_group_name
+        self.owner_id = owner_id
+        self.partition_manager = partition_manager
+        self.ownership_timeout = ownership_timeout
+
+    async def claim_ownership(self):
```

Comment:
Missing type annotations...

Reply:
This is internal stuff, so I guess wo don't need type annotations? The module name starts with _ even though this method doesn't.